src/Fmt_ast.ml

```diff
@@ -3453,110 +3562,116 @@ and fmt_structure_item c ~sep ~last:last_item ?ext ?(use_file= false)
-      (* recognize and undo the pattern of code introduced by
-         ocaml/ocaml@fd0dc6a0fbf73323c37a73ea7e8ffc150059d6ff to fix
-         https://caml.inria.fr/mantis/view.php?id=7344 *)
-      | ( Ppat_constraint
-            ( ({ppat_desc= Ppat_var _} as pat)
-            , {ptyp_desc= Ptyp_poly ([], typ1)} )
-        , Pexp_constraint (_, typ2) )
-        when Poly.equal typ1 typ2 ->
-          sub_pat ~ctx:(Pat pvb_pat) pat
-      | _ -> sub_pat ~ctx pvb_pat
-    in
-    let pat_is_extension {ppat_desc} =
-      match ppat_desc with Ppat_extension _ -> true | _ -> false
-    in
-    let ({ast= body} as xbody) = sub_exp ~ctx pvb_expr in
-    if
-      (not (List.is_empty xbody.ast.pexp_attributes))
-      || pat_is_extension pat
-    then (xpat, [], None, xbody)
-    else
-      let sugar_polynewtype pat body =
-        let ctx = Pat pat in
-        match pat.ppat_desc with
-        | Ppat_constraint (pat, {ptyp_desc= Ptyp_poly (pvars, _)}) ->
-            let rec sugar_polynewtype_ xpat pvars0 pvars body =
-              let ctx = Exp body in
-              match (pvars, body.pexp_desc) with
-              | [], Pexp_constraint (exp, typ) ->
-                  Some (xpat, pvars0, sub_typ ~ctx typ, sub_exp ~ctx exp)
-              | {txt= pvar} :: pvars, Pexp_newtype ({txt= nvar}, exp)
-                when String.equal pvar nvar ->
-                  sugar_polynewtype_ xpat pvars0 pvars exp
-              | _ -> None
-            in
-            sugar_polynewtype_ (sub_pat ~ctx pat) pvars pvars body
-        | _ -> None
-      in
-      match sugar_polynewtype pat body with
-      (* format
-       *    let f: 'r 's. 'r 's t = fun (type r) -> fun (type s) -> (e : r s t)
-       * as let f: type r s. r s t = e *)
-      | Some (xpat, pvars, xtyp, xbody) ->
-          let fmt_cstr =
-            fmt "@ : type "
-            $ list pvars " " (fun {txt} -> str txt)
-            $ fmt ".@ " $ fmt_core_type c xtyp
-          in
-          (xpat, [], Some fmt_cstr, xbody)
-      | None ->
-          let xpat =
-            match xpat.ast.ppat_desc with
-            | Ppat_constraint (p, {ptyp_desc= Ptyp_poly ([], _)}) ->
-                sub_pat ~ctx:xpat.ctx p
-            | _ -> xpat
-          in
-          let xargs, ({ast= body} as xbody) =
-            sugar_fun c (Some pat) xbody
-          in
-          let fmt_cstr, xbody =
-            let ctx = Exp body in
-            match (body.pexp_desc, pat.ppat_desc) with
-            | ( Pexp_constraint
-                  ({pexp_desc= Pexp_pack _}, {ptyp_desc= Ptyp_package _})
-              , _ )
-             |Pexp_constraint _, Ppat_constraint _ ->
-                (None, xbody)
-            | Pexp_constraint (exp, typ), _ ->
-                ( Some
-                    (fmt "@ " $ fmt_core_type c ~pro:":" (sub_typ ~ctx typ))
-                , sub_exp ~ctx exp )
-            | _ -> (None, xbody)
+  maybe_disabled c pvb_attributes pvb_loc (fun () ->
```

Comment:
This will format better using `maybe_disabled c pvb_attributes pvb_loc @@ (fun () ->`

Reply:
indeed